hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -417,9 +451,12 @@ private SCMNodeStat getNodeStatInternal(DatanodeDetails datanodeDetails) {
 
   @Override
   public Map<String, Integer> getNodeCount() {
+    // TODO - This does not consider decom, maint etc.
     Map<String, Integer> nodeCountMap = new HashMap<String, Integer>();
```

Comment:
Why not <NodeState, Integer>? It makes it easier to consume for the caller in my opinion.

Reply:
The existing code had Map<String, Integer>, but I agree it would be better with <NodeState, Integer> or <NodeStatus, Integer>. I plan to leave this as is for now, as this method is used only for JMX right now, and I plan to split that out into a separate change via HDDS-2113 as there are some open questions there.